### PR TITLE
feat(ubuntu): added support for Ubuntu ESM versions

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Ubuntu has [ESM program](https://ubuntu.com/security/esm).
ESM status information is stored in the`var/lib/ubuntu-advantage/status.json` file.
- added parsing `status.json` file.
- added merging Version OS and ESM status.
- added tests for Ubuntu ESM.

## Blockers
- [ ] aquasecurity/trivy-db/pull/200

### CODE REVIEW

File: analyzer/os/ubuntu/ubuntu.go [hunk 1]
```diff
@@ -18,9 +20,22 @@ func init() {
...
+	ubuntuConfFilePath = "etc/lsb-release"
+	esmConfFilePath    = "var/lib/ubuntu-advantage/status.json"
+	esmServiceName     = "esm-infra"
+	esmStatusEnabled   = "enabled"
+	esmVersionSuffix   = "ESM"
+)
 
-var requiredFiles = []string{"etc/lsb-release"}
+var (
+	osVersion     = ""
```
Comment (AndreyLevchenko) [inline analyzer/os/ubuntu/ubuntu.go:33]: is package scope for `osVersion` correct? Can it be used more than once in one fanal invocation?

File: analyzer/os/ubuntu/ubuntu.go [hunk 2]
```diff
@@ -57,3 +94,30 @@ func (a ubuntuOSAnalyzer) Type() analyzer.Type {
...
+func esmEnabled(config string) bool {
+	st := status{}
+
+	err := json.Unmarshal([]byte(config), &st)
+	if err != nil {
+		return false
+	}
+
+	for _, s := range st.Services { // Find ESM Service
+		if s.Name == esmServiceName {
```
Comment (AndreyLevchenko) [inline analyzer/os/ubuntu/ubuntu.go:116]: can single `if` statement be used here?

File: analyzer/os/ubuntu/ubuntu_test.go [hunk 3]
```diff
@@ -14,36 +15,91 @@ import (
 
 func Test_ubuntuOSAnalyzer_Analyze(t *testing.T) {
 	tests := []struct {
-		name      string
-		inputFile string
-		want      *analyzer.AnalysisResult
-		wantErr   string
+		name            string
+		firstInputFile  string
```
Comment (AndreyLevchenko) [inline analyzer/os/ubuntu/ubuntu_test.go:19]: make it a collection. (to handle a case when third file is required)

Comment (DmitriyLewen) [reply analyzer/os/ubuntu/ubuntu_test.go]: Done

File: analyzer/os/ubuntu/ubuntu.go (hunk 2 above)
Comment (DmitriyLewen) [reply analyzer/os/ubuntu/ubuntu.go]: Done

File: analyzer/os/ubuntu/ubuntu.go (hunk 1 above)
Comment (DmitriyLewen) [reply analyzer/os/ubuntu/ubuntu.go:33]: `osVersion` value is correctly preserved for all calls to `Analyze` function.

Comment (DmitriyLewen) [reply analyzer/os/ubuntu/ubuntu.go:33]: I was wrong. 
When scanning more than 1 image, we got the wrong `osVersion`.
I used `context` to save the osVersion parts for each image. 
Can you take a look, when you will time.

**Pull request closed without merging**